[browserperfdash-benchmark] Support using a streaming http response when receiving the status update from the remote dashboard

clopez opened the pull request:
#### 7cc16166ce74d1b1697ae8498200ad7b2d2d5a9c
<pre>
[browserperfdash-benchmark] Support using a streaming http response when receiving the status update from the remote dashboard
<a href="https://bugs.webkit.org/show_bug.cgi?id=275393">https://bugs.webkit.org/show_bug.cgi?id=275393</a>

Reviewed by Nikolas Zimmermann.

The browserperfdash server now sends the reply back to the worker using a streaming http response
that each second outputs a dot (&apos;.&apos;) to indicate that the processing of the data is ongoing on
the remote server. This keeps the connection open on the intermediate gateway.
At the end of the reply it sends the status code and the message with the result of the operation.
Refs: <a href="https://github.com/Igalia/browserperfdash/commit/0193bf5">https://github.com/Igalia/browserperfdash/commit/0193bf5</a>

Implement on the runner support for this new type of reply.

* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
(BrowserPerfDashRunner):
(BrowserPerfDashRunner._read_stream_response):
(BrowserPerfDashRunner._upload_result):

Canonical link: <a href="https://commits.webkit.org/280196@main">https://commits.webkit.org/280196@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ceff6cc64a45ead8c0805ea4d352d2446eb5b30d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54997 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34454 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7590 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58275 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5728 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57297 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42004 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5760 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44542 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3898 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57092 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32538 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47647 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25668 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/54823 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29324 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4991 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3869 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51203 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5212 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59866 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30264 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5372 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51961 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31394 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47723 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51409 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32416 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8276 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31185 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann reviewed (approved):
Hard to review without test data, etc.. however, since I trust you, r=me :-)

clopez committed:
[browserperfdash-benchmark] Support using a streaming http response when receiving the status update from the remote dashboard
https://bugs.webkit.org/show_bug.cgi?id=275393

Reviewed by Nikolas Zimmermann.

The browserperfdash server now sends the reply back to the worker using a streaming http response
that each second outputs a dot ('.') to indicate that the processing of the data is ongoing on
the remote server. This keeps the connection open on the intermediate gateway.
At the end of the reply it sends the status code and the message with the result of the operation.
Refs: https://github.com/Igalia/browserperfdash/commit/0193bf5

Implement on the runner support for this new type of reply.

* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_runner.py:
(BrowserPerfDashRunner):
(BrowserPerfDashRunner._read_stream_response):
(BrowserPerfDashRunner._upload_result):

Canonical link: https://commits.webkit.org/280196@main
